### PR TITLE
Remote datasets: Include images of the official documentation in API reference

### LINKED ISSUE #2325: Default CPTs for GMT remote datasets
Most GMT remote datasets have a default CPT. When plotting a dataset without specifying a CPT, the default CPT will be used. For example, the default CPT for `earth_age` dataset is `@earth_age.cpt`. 

The following command plots the `earth_age_01d_g.nc` grid using the default `@earth_age.cpt` file.
```
gmt grdimage @earth_age_01d_g -png map
```
map

In the following PyGMT script, a grid file name is directly passed to `Figure.grdimage`. This script produces the indentical image:
```python
import pygmt
fig = pygmt.Figure()
fig.grdimage(grid="@earth_age_01d_g")
fig.colorbar(frame=True)
fig.savefig("map.png")
```
map

However, if we load the dataset into a xarray.DataArray and pass it to `Figure.grdimage`, the default CPT changes to "turbo" (GMT's default CPT for any grid data).
```python
import pygmt
from pygmt.datasets import load_earth_age

grid = load_earth_age()
fig = pygmt.Figure()
fig.grdimage(grid=grid)
fig.colorbar()
fig.savefig("map.png")
```
map

It would be good if PyGMT can use the specific CPT when a CPT is not specified, although it's technically difficult. At least we should document the default CPT for each dataset and add an example showing how to use the default dataset CPT.

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to include images of the remote datasets in the API reference by adding the links to the images showen at https://www.generic-mapping-tools.org/remote-datasets/.

Fixes #2325 

**Preview**: 
- https://pygmt-dev--2728.org.readthedocs.build/en/2728/api/index.html#datasets
- https://pygmt-dev--2728.org.readthedocs.build/en/2728/api/generated/pygmt.datasets.load_earth_magnetic_anomaly.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by yvonnefroehlich** [comment]
Some questions:

- For now, I placed the images just above the `Parameters` section. Or do people prefer another position?
- There are two magnetic datasets: `EMAG2` (https://www.generic-mapping-tools.org/remote-datasets/earth-mag.html) and `WDMAM` (https://www.generic-mapping-tools.org/remote-datasets/earth-wdmam.html). Do we want to add both figures side by side in a tabular manner?
- Do we also want to add the image for `earth_mask` (https://www.generic-mapping-tools.org/remote-datasets/earth-mask.html) even though the used categorical colormap is not defined as a GMT default colormap?

File: pygmt/datasets/earth_age.py [hunk 1]
```diff
@@ -36,6 +36,12 @@ def load_earth_age(resolution="01d", region=None, registration=None):
     Refer to :gmt-datasets:`earth-age.html` for more details about available
     datasets, including version information and references.
 
+    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_age.png # noqa: W505
+    :width: 700 px
+    :align: center
+
+    Earth seafloor crustal age dataset.
```
Comment (seisman) [inline pygmt/datasets/earth_age.py:43]: Need indentations to make it work:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_age.png # noqa: W505
        :width: 700 px
        :align: center

        Earth seafloor crustal age dataset.
```

**Comment by seisman** [comment]
> * For now, I placed the images just above the `Parameters` section. Or do people prefer another position? Maybe above the `Refer to xyz` sentence?

I prefer to put the images at the top, just below the "Load xxx dataset in various resolutions" sentence, because the images are more eye-catching than the long description.

I also prefer to make the images slightly smaller. I usually use a relative width like `width: 80%` which is 80% of the content width rather than an absolute width like `width: 700px`.

> * There are two magnetic datasets: `EMAG2` (https://www.generic-mapping-tools.org/remote-datasets/earth-mag.html) and `WDMAM` (https://www.generic-mapping-tools.org/remote-datasets/earth-wdmam.html). Do we want to add both figures side by side in a tabular manner?

They look quite different in some places. A side-by-side table sounds a good idea.

> * Do we also want to add the image for `earth_mask` (https://www.generic-mapping-tools.org/remote-datasets/earth-mask.html) even though the used categorical colormap is not defined as a GMT default colormap?

I think the colormap really doesn't matter much for earth_mask. 
Unlike other images, the earth_mask image has a frame and annotations, which doesn't look good. I think we should open an upstream issue to fix it.

**Comment by yvonnefroehlich** [comment]
> I prefer to put the images at the top, just below the "Load xxx dataset in various resolutions" sentence, because the images are more eye-catching than the long description.

That's fine with me and is done in commit https://github.com/GenericMappingTools/pygmt/pull/2728/commits/fdfa59950a0dd04e7dc75a736701254f936a07d3.

> I also prefer to make the images slightly smaller. I usually use a relative width like `width: 80%` which is 80% of the content width rather than an absolute width like `width: 700px`.

Using a relative width is also fine with me.
However, there seem to be no big difference in the size of the images between using `700 px` and `80 %`.

> > * Do we also want to add the image for `earth_mask` (https://www.generic-mapping-tools.org/remote-datasets/earth-mask.html) even though the used categorical colormap is not defined as a GMT default colormap?
> 
> I think the colormap really doesn't matter much for earth_mask. Unlike other images, the earth_mask image has a frame and annotations, which doesn't look good. I think we should open an upstream issue to fix it.

Yeah, I was also wondering about the annotated frame and think this image should be consistent with the images for the other datasets.

File: pygmt/datasets/earth_magnetic_anomaly.py [hunk 2]
```diff
@@ -18,6 +18,17 @@ def load_earth_magnetic_anomaly(
     r"""
     Load the Earth magnetic anomaly datasets in various resolutions.
 
+    +-----------------------------------+-----------------------------------+
```
Comment (seisman) [inline pygmt/datasets/earth_magnetic_anomaly.py:21]: Better to use the `list-table` syntax (https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table).

The following may work:
```
.. list-table:: 
   :widths: 50 50
   :header-rows: 1

   * - Global Earth Magnetic Anomaly Model (EMAG2)
     - World Digital Magnetic Anomaly Map (WDMAM)
   * - .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_mag.jpg
     - .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_wdmam.jpg
```

Comment (yvonnefroehlich) [reply pygmt/datasets/earth_magnetic_anomaly.py]: Thanks! I made this change in commit https://github.com/GenericMappingTools/pygmt/pull/2728/commits/5b17e6b4a2ae7c93dec2db0ee5f124e825e4bae4.

File: pygmt/datasets/earth_magnetic_anomaly.py [hunk 3]
```diff
@@ -18,6 +18,19 @@ def load_earth_magnetic_anomaly(
...
+       :header-rows: 1
+
+       * - Global Earth Magnetic Anomaly Model (EMAG2)
+         - World Digital Magnetic Anomaly Map (WDMAM)
+       * - .. figure::
+              https://www.generic-mapping-tools.org/
+              remote-datasets/_images/GMT_mag.jpg
+         - .. figure::
+              https://www.generic-mapping-tools.org/
+              remote-datasets/_images/GMT_wdmam.jpg
```
Comment (seisman) [inline pygmt/datasets/earth_magnetic_anomaly.py:32]: Does this work?
```suggestion
       * - .. figure::
              https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_mag.jpg # noqa: W505
         - .. figure::
              https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_wdmam.jpg # noqa: W505
```

Comment (yvonnefroehlich) [reply pygmt/datasets/earth_magnetic_anomaly.py:32]: Locally, the Style Checks fail (even when `#noqa w505` is used):
```
************* Module pygmt.datasets.earth_magnetic_anomaly
pygmt/datasets/earth_magnetic_anomaly.py:30:0: C0301: Line too long (102/100) (line-too-long)
```

Comment (seisman) [reply pygmt/datasets/earth_magnetic_anomaly.py:32]: What's your `flakeheaven` and `pylint` versions? 

I have 
```
$ flakeheaven --version
FlakeHeaven 3.3.0
Flake8    4.0.1

$ pylint --version
pylint 3.0.1
astroid 3.0.0
Python 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
```
and it seems working for me.

**Comment by seisman** [comment]
The following warning is from pylint:
```
pygmt/datasets/earth_magnetic_anomaly.py:30:0: C0301: Line too long (102/100) (line-too-long)
```
and the following warning is from pycodestyle:
```
pygmt/datasets/earth_magnetic_anomaly.py:28:80: W505 doc line too long (119 > 79 characters) [pycodestyle]
```
It seems we have two tools that check line length. Maybe we should disable one?

BTW, the following code can disable both warnings but itself is too long: 
```
# pylint: disable=line-too-long # noqa: W505
```

**Comment by yvonnefroehlich** [comment]
> The following warning is from pylint:
> 
> ```
> pygmt/datasets/earth_magnetic_anomaly.py:30:0: C0301: Line too long (102/100) (line-too-long)
> ```
> 
> and the following warning is from pycodestyle:
> 
> ```
> pygmt/datasets/earth_magnetic_anomaly.py:28:80: W505 doc line too long (119 > 79 characters) [pycodestyle]
> ```
> 
> It seems we have two tools that check line length. Maybe we should disable one?
> 
> BTW, the following code can disable both warnings but itself is too long:
> 
> ```
> # pylint: disable=line-too-long # noqa: W505
> ```

If we us a tool setting a limit of `< 79` signs, I do not see the need or sense of having an additional tool for checking for `< 100` signs. So, if we would decide to not use `pylint`'s limit `< 100` anymore, we would only have the limit `< 79` from `pycodestyle` and this warning could be disabled via `# noqa: W505`. Then it should be possible to have the link in one line, correct?

**Comment by seisman** [comment]
> > The following warning is from pylint:
> > ```
> > pygmt/datasets/earth_magnetic_anomaly.py:30:0: C0301: Line too long (102/100) (line-too-long)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and the following warning is from pycodestyle:
> > ```
> > pygmt/datasets/earth_magnetic_anomaly.py:28:80: W505 doc line too long (119 > 79 characters) [pycodestyle]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It seems we have two tools that check line length. Maybe we should disable one?
> > BTW, the following code can disable both warnings but itself is too long:
> > ```
> > # pylint: disable=line-too-long # noqa: W505
> > ```
> 
> If we us a tool setting a limit of `< 79` signs, I do not see the need or sense of having an additional tool for checking for `< 100` signs. So, if we would decide to not use `pylint`'s limit `< 100` anymore, we would only have the limit `< 79` from `pycodestyle` and this warning could be disabled via `# noqa: W505`. Then it should be possible to have the link in one line, correct?

Done in PR https://github.com/GenericMappingTools/pygmt/pull/2735.

**Comment by seisman** [comment]
PR #2735 has been merged, now it's possible to write like:
```
.. figure:: https://long/path/to/image.png # noqa: W505
```

File: pygmt/datasets/earth_age.py [hunk 4]
```diff
@@ -15,6 +15,13 @@ def load_earth_age(resolution="01d", region=None, registration=None):
     r"""
     Load the Earth seafloor crustal age dataset in various resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_age.png # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_age.py:19]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_age.png # noqa: W505
```

File: pygmt/datasets/earth_free_air_anomaly.py [hunk 5]
```diff
@@ -16,6 +16,13 @@ def load_earth_free_air_anomaly(resolution="01d", region=None, registration=None
     Load the IGPP Global Earth Free-Air Anomaly datatset in various
     resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_faa.jpg # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_free_air_anomaly.py:20]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_faa.jpg # noqa: W505
```

File: pygmt/datasets/earth_geoid.py [hunk 6]
```diff
@@ -15,6 +15,13 @@ def load_earth_geoid(resolution="01d", region=None, registration=None):
     r"""
     Load the EGM2008 Global Earth Geoid dataset in various resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_geoid.jpg # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_geoid.py:19]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_geoid.jpg # noqa: W505
```

File: pygmt/datasets/earth_mask.py [hunk 7]
```diff
@@ -15,6 +15,13 @@ def load_earth_mask(resolution="01d", region=None, registration=None):
     r"""
     Load the GSHHG Global Earth Mask dataset in various resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_earthmask.png # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_mask.py:19]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_earthmask.png # noqa: W505
```

File: pygmt/datasets/earth_relief.py [hunk 8]
```diff
@@ -23,6 +23,13 @@ def load_earth_relief(
     Load the Earth relief datasets (topography and bathymetry) in various
     resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_gebcosi.jpg # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_relief.py:27]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_gebcosi.jpg # noqa: W505
```

File: pygmt/datasets/earth_vertical_gravity_gradient.py [hunk 9]
```diff
@@ -18,6 +18,13 @@ def load_earth_vertical_gravity_gradient(
     Load the IGPP Global Earth Vertical Gravity Gradient dataset in various
     resolutions.
 
+    .. figure::
+       https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_vgg.jpg # noqa: W505
```
Comment (seisman) [inline pygmt/datasets/earth_vertical_gravity_gradient.py:22]:
```suggestion
    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_vgg.jpg # noqa: W505
```

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
@yvonnefroehlich See https://github.com/GenericMappingTools/remote-datasets/pull/89 for the upstream changes of image names.

**Pull request merged**